### PR TITLE
Added DENO_DEPLOYMENT_ID env variable

### PR DESCRIPTION
**Pull request by cdoremus**
Added `DENO_DEPLOYMENT_ID` environmental variable needed for Docker deployment of a Deno
Fresh app for caching to work properly (see https://fresh.deno.dev/docs/concepts/deployment#docker).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,6 +251,22 @@ services:
       - "8000:8000"
 ```
 
+The environmental variable values are pulled from the `.env` file.
```
Comment (lambtron) [inline README.md:254]:
```suggestion
The values of the environmental variables are pulled from the `.env` file.
```

**Comment by lambtron** [comment]
am i understanding the directions correctly RE: docker deployment that for each deployment, you have to run

```
git rev-parse HEAD
```

and then copy and paste that into `.env` file as `DENO_DEPLOYMENT_ID`?

maybe we suggest using GitHub Actions, in which it will grab the `git rev-parse HEAD` and assign it to `DENO_DEPLOYMENT_ID` before it builds/ tags the Docker image, and deploys it to whichever VPS.

what do you think?

**Comment by cdoremus** [comment]
> maybe we suggest using GitHub Actions, in which it will grab the `git rev-parse HEAD` and assign it to `DENO_DEPLOYMENT_ID` before it builds/ tags the Docker image, and deploys it to whichever VPS.
> 
> what do you think?

As we talked about on Discord, this is something that we would implement in the future.

File: README.md [hunk 2]
```diff
@@ -255,6 +256,22 @@ services:
...
+that can be obtained from the following command run in the repo's root folder.
+Copy its results into the value of `DENO_DEPLOYMENT_ID`. Make sure you update
+this value every time you run a new docker compose build.
+
+```sh
+# get the SHA1 commit hash of the current branch
+git rev-parse HEAD
+```
+
+Make sure this command is run and value upgraded
```
Comment (lambtron) [inline README.md:273]:
```suggestion
Make sure this command is run and value updated in your .env.
```

File: README.md [hunk 3]
```diff
@@ -255,6 +256,22 @@ services:
...
 ```
 
+The values of the environmental variables are pulled from the `.env` file.
+
+The `DENO_DEPLOYMENT_ID` variable is needed for Docker deployment of a Deno
+Fresh app for caching to work properly. Its value needs to be a token reflecting
+the build that you will be deploying. The SHA1 commit hash is a suitable token
+that can be obtained from the following command run in the repo's root folder.
+Copy its results into the value of `DENO_DEPLOYMENT_ID`. Make sure you update
+this value every time you run a new docker compose build.
```
Comment (lambtron) [inline README.md:266]:
```suggestion
The `DENO_DEPLOYMENT_ID` variable is needed for Docker deployment of a Deno
Fresh app for caching to work properly. Its value needs to be a unique id tied to the deployment. We recommend using the SHA1 commit hash, which can be obtained from the following command run in the repo's root folder:

```sh
# get the SHA1 commit hash of the current branch
git rev-parse HEAD
```

Copy its results into the value of `DENO_DEPLOYMENT_ID` in your `.env` file.
```

File: README.md (hunk 2 above)
Comment (lambtron) [inline README.md:273]: we can prob remove this part given the re-organization of the information in the paragraph prior

**Comment by lambtron** [comment]
LGTM after those changes!

**Review by lambtron — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (cdoremus) [reply README.md:273]: I agree that the last line (273) needs to be removed, but the `git rev-parse HEAD` command should remain as the previous paragraph points to it. Don't you agree?

**Comment by iuioiua** [comment]
Could you please fix the CI errors?

**Comment by iuioiua** [comment]
Is this PR ready to merge?

**Comment by cdoremus** [comment]
> Is this PR ready to merge?

Yes

**Comment by cdoremus** [comment]
> Could you please fix the CI errors?

Done in e180b16

**Comment by iuioiua** [comment]
Thanks for this, @lambtron and @cdoremus!

**Pull request merged**